Managers/SettingsMan.cpp

```diff
@@ -1,637 +1,497 @@
-    writer << m_PrintDebugInfo;
-	writer.NewProperty("RecommendedMOIDCount");
-	writer << m_RecommendedMOIDCount;
-    writer.NewProperty("SoundPanningEffectStrength");
-    writer << m_SoundPanningEffectStrength;
-
-    //////////////////////////////////////////////////
-    //TODO These need to be removed when our soundscape is sorted out. They're only here temporarily to allow for easier tweaking by pawnis.
-    writer.NewProperty("ListenerZOffset");
-    writer << m_ListenerZOffset;
-    writer.NewProperty("MinimumDistanceForPanning");
-    writer << m_MinimumDistanceForPanning;
-    //////////////////////////////////////////////////
-
-	writer.NewProperty("PlayerNetworkName");
-	if (m_PlayerNetworkName == "")
-		writer << "Dummy";
-	else
+	const std::string SettingsMan::c_ClassName = "Settings";
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void SettingsMan::Clear() {
+		m_SettingsNeedOverwrite = false;
+
+		m_ForceVirtualFullScreenGfxDriver = false;
+		m_ForceOverlayedWindowGfxDriver = false;
+		m_ForceNonOverlayedWindowGfxDriver = false;
+
+		m_SoundPanningEffectStrength = 0.6F;
+		//////////////////////////////////////////////////
+		//TODO These need to be removed when our soundscape is sorted out. They're only here temporarily to allow for easier tweaking by pawnis.
+		m_ListenerZOffset = 0;
+		m_MinimumDistanceForPanning = 50.0F;
+		//////////////////////////////////////////////////
+
+		m_FlashOnBrainDamage = true;
+		m_BlipOnRevealUnseen = true;
+		m_EndlessMode = false;
+		m_EnableHats = false;
+
+		m_NetworkServerAddress = "127.0.0.1:8000";
+		m_PlayerNetworkName = "Dummy";
+		m_UseNATService = false;
+		m_NATServiceAddress = "127.0.0.1:61111";
+		m_NATServerName = "DefaultServerName";
+		m_NATServerPassword = "DefaultServerPassword";
+		m_ClientInputFps = 30;
+		m_ServerTransmitAsBoxes = true;
+		m_ServerBoxWidth = 32;
+		m_ServerBoxHeight = 44;
+		m_ServerUseHighCompression = true;
+		m_ServerUseFastCompression = false;
+		m_ServerHighCompressionLevel = 10;
+		m_ServerFastAccelerationFactor = 1;
+		m_ServerUseInterlacing = false;
+		m_ServerEncodingFps = 30;
+		m_ServerSleepWhenIdle = false;
+		m_ServerSimSleepWhenIdle = false;
+
+		m_AllowSavingToBase = false;
+		m_ShowForeignItems = true;
+		m_ShowMetaScenes = false;
+
+		m_RecommendedMOIDCount = 240;
+		m_PreciseCollisions = true;
+
+		m_PlayIntro = true;
+		m_ToolTips = true;
+		m_DisableLoadingScreen = true;
+		m_LoadingScreenReportPrecision = 100;
+		m_MenuTransitionSpeed = 1.0F;
+		m_PrintDebugInfo = false;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	int SettingsMan::Create(Reader &reader, bool checkType, bool doCreate) {
+		if (reader.IsOK()) {
+			return Serializable::Create(reader, checkType);
+		}
+
+		// Couldn't find the settings file, so create a new one with all the good defaults!
```

Comment:
It's correct because it only runs if `reader.IsOK()` returns false, meaning it couldn't find the settings file.
I deleted the other comments but i'll keep this one.

Reply:
Oh yeah, not sure what I was thinking. Maybe I misread it or something? Anyway, that sounds good to me.